### PR TITLE
Consolidate authentication config in CRD

### PR DESCRIPTION
**Pull request by maltesander**
# Description

fixes https://github.com/stackabletech/airflow-operator/issues/298

test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/airflow-operator-it-custom/94/

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator-binary/src/config.rs [hunk 1]
```diff
@@ -16,44 +16,46 @@ pub const PYTHON_IMPORTS: &[&str] = &[
...
 fn append_authentication_config(
     config: &mut BTreeMap<String, String>,
-    authentication_config: &AirflowClusterAuthenticationConfig,
-    authentication_class: &AuthenticationClass,
+    authentication_config: &Vec<AirflowAuthenticationConfigResolved>,
 ) {
-    if let AuthenticationClassProvider::Ldap(ldap) = &authentication_class.spec.provider {
-        append_ldap_config(config, ldap);
-    }
+    // TODO: we make sure in crd/src/authentication.rs that currently there is only one
```
Comment (adwk67) [inline rust/operator-binary/src/config.rs:36]: Would this comment be more visible if we moved it to AirflowClusterConfig?

File: rust/crd/src/authentication.rs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    /// Retrieve all provided `AuthenticationClass` references.
+    pub async fn resolve(
+        &self,
+        client: &Client,
+    ) -> Result<Vec<AirflowAuthenticationConfigResolved>> {
+        let mut resolved = vec![];
+
+        // TODO: adapt if multiple authentication classes are supported by airflow.
+        //    This is currently not possible due to the Flask App Builder not supporting it.
```
Comment (adwk67) [inline rust/crd/src/authentication.rs:108]: Maybe add a comment re. size<=1 to AirflowClusterConfig?

**Review by adwk67 — APPROVED** [review]
lgtm!

File: rust/operator-binary/src/config.rs (hunk 1 above)
Comment (maltesander) [reply rust/operator-binary/src/config.rs:36]: I like to keep the todos, but ill add the size limit comment to AirflowClusterConfig for the CRD description?

Comment (adwk67) [reply rust/operator-binary/src/config.rs:36]: Agreed.

File: rust/crd/src/authentication.rs (hunk 2 above)
Comment (adwk67) [reply rust/crd/src/authentication.rs:108]: See comment above.

**Review by adwk67 — DISMISSED** [review]
LGTM!

**Comment by maltesander** [comment]
bors merge

**Comment by maltesander** [comment]
bors retry

**Review by adwk67 — APPROVED** [review]

**Pull request merged**